### PR TITLE
Feature categories

### LINKED ISSUE #152: Add categories
A category - or path - is any area of programming knowledge that Mumuki covers, like 'introduction to programming', 'functional programming' or 'metaprogramming' .

It can be available in one or more programming languages, and one or more locales, and for each programming language, locale and category there are one or more associated guides. 

The intereseting part is that, given a language and locale a category allows to point to a suggested starter guide.

### PR DESCRIPTION
**Pull request by flbulgarelli**
Adding Categories

Fixes #152

### CODE REVIEW

**Comment by flbulgarelli** [comment]
Preview it here: http://stg.mumuki.io/es/categories

**Comment by flbulgarelli** [comment]
@faloi @aguspina @PalumboN I would like to have some feedback about this feature

File: app/views/categories/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+          <h3><%= it.name %></h3>
+
+          <hr>
+
+          <p>
+            <%= it.description %>
+          </p>
+
+          <p>
+            <strong>Disponible en lo siguientes lenguajes</strong>
```
Comment (faloi) [inline app/views/categories/index.html.erb]: You have a typo here, it should be "en loS siguientes lenguajes"

**Comment by faloi** [comment]
- Some strings are hardcoded into the html, I assume that you'll internationalize them later.
- Why is Gobstones missing in the preview? :cry: 
- Where is the category specified? I think the language should have a default category, overridable in a per guide basis (I'm thinking on multi-paradigm languages here).

**Comment by PalumboN** [comment]
Like. This begins the idea of _courses_.
I would change the **question** "_¿Qué te gustaría aprender?_" by an **invitation**: "_Aprendé a programar ahora mismo. Es fácil y rápido._" or something like that. I dont know, anyway dont care, the designer probably change it later :P

**Comment by flbulgarelli** [comment]
@faloi The model is a bit different.

What I am proposing here is to have categories (like OO, fundamentals, metaprogramming, etc). And then "staring points" - I am ot sure it is the best name, any other suggestions? - which tell the initial guide for a language and category. 

There may be no starting point for a languge and category - for example there would be no starting point for Haskell and object oriented programming - but at most one starting point for that pair - (language_id, category_id) should have an unique index in starting_points table.

That means that there is no direct association between a language and a paradigm/category, but always through an staring point. That is how multiparadigm is supported: there could be a staring point for JS in both OO and functional programming. 

Categories and starting points should be created manually through the admin. Please notice that Guides do not have an associated category right now, but they have suggested guides: we are modeling and storing the path as a linked structure instead of a plain instead of a collections of guides in a category.  

Finally, there are no staring points without an initial guide - the idea of starting_point is to have a clean path so that entering the platform for the first time is easier. That is why Gobstones is not listed yet - there is no guide.

**Comment by faloi** [comment]
I don't deserve to live in the same world as you.

Let me see the code again with this in mind.

**Comment by aguspina** [comment]
This is amazing!

**Pull request merged**